File: leetcode/00056_merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        idx = 0
        for i in range(1, len(intervals)):
            if intervals[i][0] <= intervals[idx][1]:
                intervals[idx][1] = max(intervals[idx][1], intervals[i][1])
            else:
                idx += 1
                intervals[idx] = intervals[i]
        return intervals[:idx+1]

File: leetcode/00056_merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>& b){
            return (a[0] < b[0]);
        });
        vector<vector<int>> ret{intervals[0]};
        for (int i = 1; i < intervals.size(); i++) {
            if (intervals[i][0] <= ret.back()[1]) {
                ret.back()[1] = max(ret.back()[1], intervals[i][1]);
            } else {
                ret.push_back(intervals[i]);
            }
        }
        return ret;
    }
};
